src/helpers.js

```diff
@@ -1,5 +1,12 @@
 export function wait(timeInMs) {
-    return new Promise(function(resolve) {
-        return setTimeout(resolve, timeInMs);
+  let timeoutHandle;
+
+  const promise = () => {
+    return new Promise(function (resolve) {
+      timeoutHandle = setTimeout(resolve, timeInMs);
+      return timeoutHandle;
     });
-}
\ No newline at end of file
+  };
+
+  return { promise, cancel: () => clearTimeout(timeoutHandle) };
```

Comment:
This is the key part of this pull request: adding the cancel method to clear the interval.